pkg/controller/release/release_controller.go

```diff
@@ -135,6 +135,13 @@ func NewController(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: controller.enqueueRelease,
 			UpdateFunc: func(oldObj, newObj interface{}) {
+				oldRel, oldOk := oldObj.(*shipper.Release)
```

Comment:
I suggest to let it bail out right away of oldOk is false. Same for newOk. Otherwise the if-statement looks overoptimistic.